### PR TITLE
Color all the things, partial patch 2

### PR DESCRIPTION
**Pull request by Carreau**
Minimal changes. 

This **does** introduce minimal inconsistency as the color themes are not yet available.

### CODE REVIEW

File: IPython/core/magics/basic.py [hunk 1]
```diff
@@ -75,6 +77,9 @@ class BasicMagics(Magics):
     These are various magics that don't fit into specific categories but that
     are all part of the base 'IPython experience'."""
 
+    def __init__(self, *args, **kwargs):
```
Comment (minrk) [inline IPython/core/magics/basic.py]: This super does nothing, right?

**Comment by minrk** [comment]
Changes here make sense, I just want to make sure I understand the goal of this partial PR:

I see pygments themes as options for `--colors` and in help output, but when I try to load one, I get:

```python
In [1]: %colors monokai
.../IPython/core/magics/basic.py:334: UserWarning: Error changing exception color schemes.
Unrecognized color scheme: monokai
Valid schemes: ['LightBG', 'NoColor', 'Linux']
```

Is there a missing `available_themes()` call somewhere?

**Comment by Carreau** [comment]
@minrk, no it's not "missing" it's just refactoring pieces by pieces. 

> This does introduce minimal inconsistency as the color themes are not yet available.

It's basically just part of https://github.com/ipython/ipython/pull/9246 , but if I want to make a fully self-contained PR  it will be hundred of lines. To @takluyver request I'm submitting smaller changes that are reviewable, though they might be not totally functional until the end.

**Comment by minrk** [comment]
@Carreau thanks for the clarification. I just wanted to know what things I should expect to work and what I shouldn't, since I can't really review it, otherwise. It would be helpful if you described a bit what you aim to accomplish in these incremental PRs.

One comment inline and a merge conflict, but otherwise :+1:.

**Pull request closed without merging**